src/Models/BaseElement.php

```diff
@@ -600,18 +600,16 @@ private function getFieldNamesExcludedFromCmsSearch(): array
      * Default way to render element in templates. Note that all blocks should
      * be rendered through their {@link ElementController} class as this
      * contains the holder styles.
-     *
-     * @return string|null HTML
      */
-    public function forTemplate($holder = true)
+    public function forTemplate($holder = true): string
     {
         $templates = $this->getRenderTemplates();
 
         if ($templates) {
             return $this->renderWith($templates);
         }
 
-        return null;
+        return '';
```

Comment:
There's no parent class enforcing non-nullable values yet - but part of the next piece of work will include adding `forTemplate()` directly on `ViewableData` where its return type will be `string`.
There's just no reason for this method to ever return `null`, in comparison with an empty string.